lib/core/request.js

```diff
@@ -271,8 +271,15 @@ class Request {
   }
 }
 
+function processHeaderValue (key, val) {
+  if (headerCharRegex.exec(val) !== null) {
+    throw new InvalidArgumentError(`invalid ${key} header`)
+  }
+  return `${key}: ${val}\r\n`
+}
+
 function processHeader (request, key, val) {
-  if (val && typeof val === 'object') {
+  if (val && !(typeof val === 'string' || typeof val === 'number' || typeof val === 'boolean' || Array.isArray(val))) {
```

Comment:
This can be simpler. Check for object and not array.

Reply:
Not sure that I follow your comment (and agree that there is likely a more simple / clean way to express that line).

The root of the original issue is that there was a `typeof val === 'object'`. The reason this is problematic is that an array will be treated the same as an object.

What I am trying to express is that the only allow header values should be allowed:
* string
* string[]
* number (cast to string) (_unofficially supported, Node's `IncomingHttpRequest` lists header values as `string` | `string[]`_)
* boolean (cast to string) (_unofficially supported, Node's `IncomingHttpRequest` lists header values as `string` | `string[]`_)

So Object's should not be allowed, except for Arrays.

How to express all of that, open to suggestions / criticism...

Reply:
(typeof val === 'object' && !Array.isArray(obj))

Reply:
Ahh, yeah... :see_no_evil: